plugins/drng/plugin.go

```diff
@@ -24,42 +21,14 @@ const PluginName = "DRNG"
 var (
 	// Plugin is the plugin instance of the DRNG plugin.
 	Plugin   = node.NewPlugin(PluginName, node.Enabled, configure, run)
-	Instance *drng.DRNG
+	instance *drng.DRNG
+	once     sync.Once
 	log      *logger.Logger
 )
 
 func configure(*node.Plugin) {
 	log = logger.NewLogger(PluginName)
-
-	// parse identities of the committee members
-	committeeMembers, err := parseCommitteeMembers()
-	if err != nil {
-		log.Warnf("Invalid %s: %s", CfgDRNGCommitteeMembers, err)
-	}
-
-	// parse distributed public key of the committee
-	var dpk []byte
-	if str := config.Node.GetString(CfgDRNGDistributedPubKey); str != "" {
-		bytes, err := hex.DecodeString(str)
-		if err != nil {
-			log.Warnf("Invalid %s: %s", CfgDRNGDistributedPubKey, err)
-		}
-		if l := len(bytes); l != cbPayload.PublicKeySize {
-			log.Warnf("Invalid %s length: %d, need %d", CfgDRNGDistributedPubKey, l, cbPayload.PublicKeySize)
-		}
-		dpk = append(dpk, bytes...)
-	}
-
-	// configure committee
-	committeeConf := &state.Committee{
-		InstanceID:    config.Node.GetUint32(CfgDRNGInstanceID),
-		Threshold:     uint8(config.Node.GetUint32(CfgDRNGThreshold)),
-		DistributedPK: dpk,
-		Identities:    committeeMembers,
-	}
-
-	Instance = drng.New(state.SetCommittee(committeeConf))
-
+	instance = Instance()
```

Comment:
This is an ineffectual assignment, as `Instance()` already sets the global variable instance.
Probably, the cleanest solution to assure that we have an instance in `configureEvents` is to do
```go
instance := Instance()
```
in that function. This makes no assumptions on `Instance()` and even works for non-singletons.